app/src/main/java/nl/tudelft/b_b_w/view/ContactsActivity.java

```diff
@@ -0,0 +1,213 @@
+package nl.tudelft.b_b_w.view;
+
+import android.app.Activity;
+import android.app.AlertDialog;
+import android.content.Context;
+import android.content.DialogInterface;
+import android.os.Bundle;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.BaseAdapter;
+import android.widget.Button;
+import android.widget.ImageView;
+import android.widget.ListAdapter;
+import android.widget.ListView;
+import android.widget.TextView;
+
+import com.jjoe64.graphview.GraphView;
+import com.jjoe64.graphview.series.DataPoint;
+import com.jjoe64.graphview.series.LineGraphSeries;
+
+import java.util.List;
+
+import nl.tudelft.b_b_w.R;
+import nl.tudelft.b_b_w.controller.BlockController;
+import nl.tudelft.b_b_w.model.Block;
+
+/**
+ * When the user wants to add a block he enters into the ContactsActivity, which contain
+ * some entry fields and a button to confirm the addition.
+ */
+public class ContactsActivity extends Activity {
+
+    private String ownerName;
+
+    /**
+     * Adapter to add the different blocks dynamically
+     */
+    private class ContactAdapter extends BaseAdapter implements ListAdapter {
+
+        //Variables which we use for getting the block information
+        private BlockController blcController;
+        Context context;
+        private final int image1 = 0;
+        private final int image2 = 1;
+        private final int image3 = 2;
+        private final int image4 = 3;
+        private final int image5 = 4;
+        //Images for displaying trust
+        private Integer images[] = {R.drawable.pic1,
+                R.drawable.pic2,
+                R.drawable.pic3,
+                R.drawable.pic4,
+                R.drawable.pic5};
+
+        /**
+         * Default constructor to initiate the Adapter
+         * @param bc BlockController which is passed on
+         * @param context Context which is passed on
+         */
+        private ContactAdapter(BlockController bc, Context context) {
+            this.context = context;
+            this.blcController = bc;
+        }
+
+        /**
+         *{@inheritDoc}
+         */
+        @Override
+        public Object getItem(int position) {
+            return blcController.getBlocks(ownerName).get(position);
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public long getItemId(int position) {
+            return 0;
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public int getCount() {
+            return blcController.getBlocks(ownerName).size();
+        }
+
+
+        /**
+         * Method to get the right image number
+         * @param trust The trust value
+         * @return Image number
+         */
+        private int getImageNo(int trust) {
+
+            if (trust >= 80) return image1;
```

Comment:
Use floor(tust/trustSize) [where trustSize=20]  instead of these hardcoded if trees